src/items.ts

```diff
@@ -1,124 +1,1166 @@
+/**
+ * items.ts
+ * All available items and their properties
+ * lgfrbcsgo & Nikolaus - October 2020
+ */
+
 export type Liter = number
 export type Minutes = number
 export type Quantity = number
 
-export type Item = Machine | Container | Ore | Other
+export type Item = Ore | Catalyst | Product | Element
 
 export enum ItemType {
-    MACHINE = "machine",
-    CONTAINER = "container",
     ORE = "ore",
-    OTHER = "product",
+    CATALYST = "catalyst",
+    PRODUCT = "product",
+    ELEMENT = "element",
 }
 
-export interface Container extends CommonProperties {
-    readonly type: ItemType.CONTAINER
+/**
+ * Properties commen to each item type
+ */
+export interface CommonProperties {
+    readonly name: string
+    readonly tier: string
+    readonly category: string
+    readonly volume: Liter
 }
 
-export interface Machine extends CommonProperties {
-    readonly type: ItemType.MACHINE
+/**
+ * Item types
+ */
+export interface Ore extends CommonProperties {
+    readonly type: ItemType.ORE
 }
 
-export interface Other extends CommonProperties {
-    readonly type: ItemType.OTHER
+export interface Catalyst extends CommonProperties {
+    readonly type: ItemType.CATALYST
 }
 
-export interface Ore extends CommonProperties {
-    readonly type: ItemType.ORE
+export interface Product extends CommonProperties {
+    readonly type: ItemType.PRODUCT
 }
 
-export interface CommonProperties {
-    readonly name: string
-    readonly volume: Liter
+export interface Element extends CommonProperties {
+    readonly type: ItemType.ELEMENT
+}
+
+/**
+ * Ore type checker
+ * @param item Item to check
+ */
+export function isOre(item: Item) {
+    return item.type === ItemType.ORE
 }
 
-export function other(name: string, volume: Liter): Other {
+/**
+ * Returns a new Ore type
+ * @param name Ore name
+ * @param volume Ore volume
+ */
+export function ore(name: string, tier: string, category: string, volume: Liter): Ore {
```

Comment:
I think we can omit `category` and `volume` here as it will always be `"Ore"` and `1`.

Reply:
Done in https://github.com/lgfrbcsgo/du-factory-generator/commit/b2ca5abf70d5f5d1f8272961ae611d9e8b5b9bf3